ts/screens/authentication/IdpLoginScreen.tsx

```diff
@@ -252,14 +336,22 @@ class IdpLoginScreen extends React.Component<Props, State> {
   }
 }
 
-const mapStateToProps = (state: GlobalState) => ({
-  loggedOutWithIdpAuth: isLoggedOutWithIdp(state.authentication)
-    ? state.authentication
-    : undefined,
-  loggedInAuth: isLoggedIn(state.authentication)
-    ? state.authentication
-    : undefined
-});
+const mapStateToProps = (state: GlobalState) => {
+  const selectedtIdp = selectedIdentityProviderSelector(state);
+  const selectedIdpTextData = selectedtIdp
```

Comment:
Since the current language could be

- the OS language if the user didn't pick a choice
- the language picked from the user in Preferences / Language

```suggestion
  const locale: Locales = preferredLanguage.map(l => l).getOrElse(
    getLocalePrimary(I18n.currentLocale())
      .map(s => s as Locales)
      .getOrElse("it")
  );
  const selectedIdpTextData = selectedtIdp
    ? idpTextDataSelector(state, selectedtIdp.id, locale)
    : undefined;
```
see the changes proposed on **idpTextDataSelector**